.ado/templates/npm-publish-steps.yml

```
@@ -26,10 +26,26 @@ steps:
 
   - script: |
       git switch $(Build.SourceBranchName)
+      echo "//registry.npmjs.org/:_authToken=$(NODE_AUTH_TOKEN)" > ~/.npmrc
       yarn nx release --skip-publish --verbose
-      yarn nx release publish --excludeTaskDependencies
     env:
       GITHUB_TOKEN: $(githubAuthToken)
+    displayName: Version Packages and Github Release
+    condition: and(succeeded(), eq(variables['publish_react_native_macos'], '1'))
+
+  - script: |
+      echo "//registry.npmjs.org/:_authToken=$(NODE_AUTH_TOKEN)" > ~/.npmrc
+      yarn nx release publish --excludeTaskDependencies
+    env:
       NODE_AUTH_TOKEN: $(npmAuthToken)
-    displayName: Version and publish packages
+    displayName: Publish packages
     condition: and(succeeded(), eq(variables['publish_react_native_macos'], '1'))
+
+  - script: |
+      if [ -f ~/.npmrc ]; then
+        rm ~/.npmrc
+      fi
```

Adding `-f` would also work:
```suggestion
      rm -f ~/.npmrc
```